### PR TITLE
Fuzzing: Add fuzzer for dev.sigstore.bundle package

### PR DESCRIPTION
**Pull request by arthurscchan**
#### Summary

Add fuzzer for dev.sigstore.bundle package

### CODE REVIEW

**Comment by arthurscchan** [comment]
@AdamKorcz

File: fuzzing/src/main/java/fuzzing/BundleFuzzer.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+
+import java.io.StringReader;
+
+import dev.sigstore.bundle.BundleFactory;
+import dev.sigstore.bundle.BundleParseException;
+import dev.sigstore.bundle.BundleVerifier;
+
+public class BundleFuzzer {
+  public static void fuzzerTestOneInput(FuzzedDataProvider data) {
```
[the remark this answers is not in the document]
Comment (AdamKorcz) [reply fuzzing/src/main/java/fuzzing/BundleFuzzer.java:27]: > Is `fuzzerTestOneInput` a name that is known by the fuzzing infra?

Yes

[the remark this answers is not in the document]
Comment (arthurscchan) [reply fuzzing/src/main/java/fuzzing/BundleFuzzer.java:27]: Indeed, it is the method name used by the fuzzing infra.

**Review by loosebazooka — DISMISSED** [review]

**Comment by loosebazooka** [comment]
This could probably use a rebase and some fixing up (would prefer rebase to merges btw).

**Review by loosebazooka — APPROVED** [review]

File: fuzzing/src/main/java/fuzzing/BundleFuzzer.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import dev.sigstore.bundle.BundleVerifier;
+import java.io.StringReader;
+
+public class BundleFuzzer {
+  public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+    try {
+      Boolean choice = data.consumeBoolean();
+      String string = data.consumeRemainingAsString();
+
+      if (choice) {
```
Comment (vlsi) [inline fuzzing/src/main/java/fuzzing/BundleFuzzer.java:30]: Could it be split in several fuzzers?

Comment (loosebazooka) [reply fuzzing/src/main/java/fuzzing/BundleFuzzer.java:30]: @arthurscchan agreed. These seem unrelated. Best to make them separate fuzzers

**Pull request merged**